### PR TITLE
Add object-shorthand transform.

### PR DESCRIPTION
**Pull request by bighuggies**

### CODE REVIEW

File: transforms/object-shorthand.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ *  identifier2,
+ *  method() {}
+ * }
+ */
+module.exports = (file, api, options) => {
+  const j = api.jscodeshift;
+  const printOptions = options.printOptions || { quote: 'single' };
+  const root = j(file.source);
+
+  const canBeSimplified = function(key, value) {
```
Comment (cpojer) [inline transforms/object-shorthand.js]: this should be an arrow function :)

File: transforms/object-shorthand.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * 
+ * var object = {
+ *  identifier,
+ *  identifier2,
+ *  method() {}
+ * }
+ */
+module.exports = (file, api, options) => {
+  const j = api.jscodeshift;
+  const printOptions = options.printOptions || { quote: 'single' };
```
Comment (cpojer) [inline transforms/object-shorthand.js]: remove the space after `{` and before `}`.

File: transforms/object-shorthand.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  const root = j(file.source);
+
+  const canBeSimplified = function(key, value) {
+    // Can be simplified if both key and value are the same identifier or if the property is a method.
+    if (key.type === 'Identifier') {
+      return (value.type === 'Identifier' && key.name === value.name) || value.type === 'FunctionExpression';
```
Comment (cpojer) [inline transforms/object-shorthand.js]: if this is more than 80 characters, it should look something like this:

```
return (
  (
    value.type === 'Identifier' &&
    key.name === value.name
  ) ||
  value.type === 'FunctionExpression'
);

this is admittedly a few more lines of code but quite a bit easier to read.

File: transforms/object-shorthand.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    // Can be simplified if the key is a string literal which is equal to the identifier name of the value.
+    if (key.type === 'Literal') {
+      return value.type === 'Identifier' && key.value === value.name;
+    }
+    
+    return false;
+  };
+
+  root
+    .find(j.Property, { method: false, shorthand: false, computed: false })
```
Comment (cpojer) [inline transforms/object-shorthand.js]: the object literal should be split up into a few lines and contain a trailing comma, like

```js
  .find(j.Property, {
    method: false,
    shorthand: false,
    computed: false,
  })

File: transforms/object-shorthand.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      
+      if (p.value.value.type === 'Identifier') {
+        p.value.shorthand = true;
+      } else if (p.value.value.type === 'FunctionExpression') {
+        p.value.method = true;
+      }
+    });
+
+  return root.toSource(printOptions);
+};
```
Comment (cpojer) [inline transforms/object-shorthand.js]: please add a newline at the end?

**Comment by cpojer** [comment]
This is awesome @bighuggies! I left a couple of code-style comments to align this with the rest of the transforms if you don't mind. Please let me know when you have made these changes because GitHub never sends an email when someone updates a PR :(

**Comment by bighuggies** [comment]
Updated @cpojer :) Let me know if I missed anything.

**Comment by cpojer** [comment]
Thank you so much! Feel free to send PRs for more codemods =)

**Comment by brentvatne** [comment]
Nice one @bighuggies!

**Pull request merged**